services/ui-src/src/components/fields/NumberField.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from "react";
+import { useFormContext } from "react-hook-form";
+// components
+import { Box } from "@chakra-ui/react";
+import { TextField } from "./TextField";
+// utils
+import {
+  CustomMasks,
+  isCustomMask,
+  makeMediaQueryClasses,
+  maskValue,
+} from "utils";
+import { InputChangeEvent, AnyObject } from "types";
+
+export const NumberField = ({
+  name,
+  label,
+  placeholder,
+  mask,
+  sxOverride,
+  nested,
+  ...props
+}: Props) => {
+  const mqClasses = makeMediaQueryClasses();
+
+  const [value, setValue] = useState(props?.hydrate || "");
+
+  // get the form context
+  const form = useFormContext();
+
+  const onBlurHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const eventValue = isCustomMask(mask)
+      ? maskValue(e.target.value, mask)
+      : e.target.value;
+    setValue(eventValue);
+    form.setValue(name, eventValue, { shouldValidate: true });
+  };
+
+  // update form data
+  const onChangeHandler = async (e: InputChangeEvent) => {
+    const { name, value } = e.target;
+    setValue(value);
+    form.setValue(name, value, { shouldValidate: true });
+  };
+
+  const errorMessage = form?.formState?.errors?.[name]?.message;
+  const nestedChildClasses = nested ? "nested ds-c-choice__checkedChild" : "";
+
+  /*
+   * Check if its not a custom mask and if the mask is defined as "currency"
+   * If its not, we dont want to use the mask prop so return undefined.
+   */
+  const useMask = mask && !isCustomMask(mask) ? mask : undefined;
```

Comment:
Where is it checking if the mask is defined as "currency"?

Also would avoid the name `useMask` because "use" is usually reserved for hooks as a standard naming convention. If you look in `utils/mask` I made some suggestions that could let us use something like `maskToUse` or `selectedMask` or `applicableMask` here like this that would return the same stuff:

```suggestion
  const maskToUse = selectedMask(mask);
```